src/app/auth/auth.component.ts

```diff
@@ -96,11 +97,22 @@ export class AuthComponent implements OnInit, OnDestroy {
     if (event.origin !== AppConfig.origin) {
       return;
     }
-    const { oauthCode } = event.data;
+
+    const { oauthCode, state } = event.data;
 
     if (!oauthCode) {
+        return;
+    }
+
+    if (!this.authService.isReturnedStateSame(state)) {
+      this.authService.changeAuthState(AuthState.NotAuthenticated);
+      this.errorHandlingService.handleError(this.errorHandlingService.stateErrorMessage);
```

Comment:
I don't think we need to display an error, or change the auth state to `NotAuthenticated`.
We can simply return out of the `onMessage` event handler if the states don't match.

This is sufficient to ensure that we don't attempt to obtain access tokens for authorisation codes that were not requested by the application.

The state parameter itself should be invisible to the user.

Reply:
Alright!

Reply:
For the auth state part, if I do not change back the state, the user will be stuck on this screen without any message 
![image](https://user-images.githubusercontent.com/66008784/110228742-74404100-7f3e-11eb-912b-e8ecf2085da5.png)
Changing back the state to NotAuthenticated will bring the user back to this screen 
![image](https://user-images.githubusercontent.com/66008784/110228753-8621e400-7f3e-11eb-833a-c383226ab6e1.png)
which means the user can try to log in again. 
Do you think it is better to keep the change of state, or should I remove the line?

Reply:
They would not be stuck; merely waiting till the auth code meant for the user (and the correct `state`) is returned from GitHub, yes?

We're simply preventing CATcher from moving forward in the authentication process with auth codes that were not requested by the user.

Reply:
My current implementation is such that the authentication session is terminated if the code is returned but the state returned with the code is not the same as the state in the existing AuthService, which means the user has to restart the process by pressing the submit button. So what I meant by “stuck” was that the current authentication session would be terminated, however the user is unable to restart the process since he is unable to go back to the “submit” button page. However I think it will be better if I continue to wait till the auth code and the correct state string instead of terminating the session, so I will change it accordingly! Thank you for the comment.

Reply:
Yes, we shouldn't terminate the authentication process if `state` doesn't match. Rather, we ignore the code and wait for a code that's accompanied with a matching `state`.